ext/node/polyfills/worker_threads.ts

```diff
@@ -126,7 +126,7 @@ class _Worker extends EventEmitter {
       }
       if (
         !(specifier.toString().endsWith(".mjs") ||
-          (pkg && pkg.exists && pkg.typ == "module"))
+          (pkg && pkg.exists && pkg.type == "module"))
```

Comment:
I'm pretty sure this is correct - if you take a look at https://github.com/denoland/deno/blob/09af1a5fef9d6c9e2c1b50484701b1cfd7c7e938/ext/node/package_json.rs#L37 you'll see that the field is actually named `typ` (to avoid clashes with Rust's `type` keyword). Arguably we should change it to `r#type` or use `#[serde(rename = "type")]`. What's concerning is that there's no failing test for this change.

Reply:
I totally missed that.  I just assumed it pulled in the nearest package.json from within TS/JS... I never considered the contents could be coming through another language layer (which was very nearsighted on my part since that's exactly what happens in **many** other area of the toolchain).

Are there any sort of "Rust-indicators" that I can keep an eye out for in the future that are signs of the JS being tied into the Rust toolchain? Or is it more of an implicit thing which files are touched by Rust and which are solely JS? Sorry for the high level of noob over here, I'm still trying to wrap my head around how Rust interacts with JS. Any tips that would help me avoid sending you guys a useless PR in the future would be much appreciated.

Thanks for your patience 🙏

Reply:
No worries, it's not immediately obvious. Any time something like `core.ops.op_<some_name>` or `await op_<some_name>` is called means that the language boundary is crossed and you should look for `op_<some_name>` is Rust files for the "op" that is called in this place.

Reply:
Thanks for clarifying that. I was reviewing the file again just now and I spotted the `core.op` call literally **_right_** above the change I made. Serves me right for not opening an issue beforehand 😅

I'll revert this now. It is concerning that no tests picked up on the change though, since I technically introduced a bug rather than eliminated one